File: data/multilang_comparison/program_cpp_array/src/func13.cpp
#include "program_cpp_array.hpp" 
Array* func13(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array102;
   if (pCounter > 0) {
      array102 = vars->data[--pCounter];
      array102->refC++;
      DEBUG_COPY(array102->id);
   } else {
      array102 = new Array();
      array102->size = 984;
      array102->refC = 1;
      array102->id = 102;
      array102->data = new unsigned int[array102->size]();
      DEBUG_NEW(array102->id);
   }
   Array_param params0;
   params0.size = 1;
   Array* data_params0[1];
   params0.data = data_params0;
   params0.data[0] = array102;
   Array* array103 = func17(&params0, loopsFactor);
   DEBUG_RETURN(array103->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   if(PATH0 & 1) {
      Array* array106;
      if (pCounter > 0) {
         array106 = vars->data[--pCounter];
         array106->refC++;
         DEBUG_COPY(array106->id);
      } else {
         array106 = new Array();
         array106->size = 672;
         array106->refC = 1;
         array106->id = 106;
         array106->data = new unsigned int[array106->size]();
         DEBUG_NEW(array106->id);
      }
      unsigned int loop32 = 0;
      unsigned int loopLimit32 = (50)/3 + 1;
      for(; loop32 < loopLimit32; loop32++) {
         for (int i = 0; i < array106->size; i++) {
             array106->data[i]--;
         }
         for (int i = 0; i < array103->size; i++) {
             if (array103->data[i] == 85) {
                 return array103;
             }
         }
      }
      Array_param params1;
      params1.size = 3;
      Array* data_params1[3];
      params1.data = data_params1;
      params1.data[0] = array102;
      params1.data[1] = array103;
      params1.data[2] = array106;
      Array* array107 = func28(&params1, get_path(), loopsFactor);
      DEBUG_RETURN(array107->id);
      if(params1.size > 0) {
      	 params1.data.clear();
      }
      array107->refC--;
      if(array107->refC == 0) {
         delete[] array107->data;
         array107->data = nullptr;
         DEBUG_FREE(array107->id);
         delete array107;
      }
      array106->refC--;
      if(array106->refC == 0) {
         delete[] array106->data;
         array106->data = nullptr;
         DEBUG_FREE(array106->id);
         delete array106;
      }
   }
   else {
      Array_param params1;
      params1.size = 2;
      Array* data_params1[2];
      params1.data = data_params1;
      params1.data[0] = array102;
      params1.data[1] = array103;
      Array* array108 = func22(&params1, get_path(), loopsFactor);
      DEBUG_RETURN(array108->id);
      if(params1.size > 0) {
      	 params1.data.clear();
      }
      array108->refC--;
      if(array108->refC == 0) {
         delete[] array108->data;
         array108->data = nullptr;
         DEBUG_FREE(array108->id);
         delete array108;
      }
   }
   array103->refC--;
   if(array103->refC == 0) {
      delete[] array103->data;
      array103->data = nullptr;
      DEBUG_FREE(array103->id);
      delete array103;
   }
   return array102;
}

File: data/multilang_comparison/program_julia/src/func13.jl
include("program_julia_head.jl")
function func13(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array102 = nothing
   if pCounter > 1
       array102 = vars.data[pCounter]
       pCounter -= 1
       array102.refC += 1
   else
       array102 = JArray(zeros(Int, 984), 984, 1, 102)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array102)
   params0.size += 1
   array103 = func17(params0, loopsFactor);
   if (PATH0 & 1) != 0
      array106 = nothing
      if pCounter > 1
          array106 = vars.data[pCounter]
          pCounter -= 1
          array106.refC += 1
      else
          array106 = JArray(zeros(Int, 672), 672, 1, 106)
      end
      loopLimit32 = (50)/3 + 1
      for i in 1:loopLimit32 
         for i in 0:(array106.size-1) 
            array106.data[i+1] -= 1
         end
         for i in 1:array103.size
             if array103.data[i] == 85
                     return array103
            end
         end
      end
      params1 = JArray_param{Int}()
      params1.size = 3
      push!(params1.data, array102)
      params1.size += 1
      push!(params1.data, array103)
      params1.size += 1
      push!(params1.data, array106)
      params1.size += 1
      array107 = func28(params1, get_path(), loopsFactor);
      array107.refC -= 1
      if array107.refC == 0
          empty!(array107.data)
          array107 = nothing
      end
      array106.refC -= 1
      if array106.refC == 0
          empty!(array106.data)
          array106 = nothing
      end
   
   else 
      params1 = JArray_param{Int}()
      params1.size = 2
      push!(params1.data, array102)
      params1.size += 1
      push!(params1.data, array103)
      params1.size += 1
      array108 = func22(params1, get_path(), loopsFactor);
      array108.refC -= 1
      if array108.refC == 0
          empty!(array108.data)
          array108 = nothing
      end
   end
   array103.refC -= 1
   if array103.refC == 0
       empty!(array103.data)
       array103 = nothing
   end
   return array102;
end

